src/main/java/org/openrewrite/apache/httpclient5/InputBufferReadAddOffsetAndLengthArguments.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Moderne Source Available License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.apache.httpclient5;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.template.Matcher;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TextComment;
+import org.openrewrite.marker.Markers;
+
+import static java.util.Objects.requireNonNull;
+
+@EqualsAndHashCode(callSuper = false)
+@Value
+public class InputBufferReadAddOffsetAndLengthArguments extends Recipe {
+    private static final String readPattern = "org.apache.hc.core5.http.nio.support.classic.SharedInputBuffer read(byte[])";
+    private static final MethodMatcher readMatcher = new MethodMatcher(readPattern);
```

Comment:
```suggestion
    private static final String READ_PATTERN = "org.apache.hc.core5.http.nio.support.classic.SharedInputBuffer read(byte[])";
    private static final MethodMatcher READ_MATCHER = new MethodMatcher(readPattern);
```

See also: https://stackoverflow.com/questions/7259687/java-naming-convention-for-static-final-variables